### PR TITLE
`Paywalls`: display purchase/restore errors

### PR DESCRIPTION
**Pull request by NachoSoto**
image

### CODE REVIEW

File: examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/paywall/PaywallScreenViewModel.kt [hunk 1]
```diff
@@ -66,16 +65,7 @@ class PaywallScreenViewModelImpl(
...
-        val value = _state.value
-        if (value is PaywallScreenState.Loaded) {
-            _state.update {
-                value.copy(
-                    dialogText = "There was an error restoring purchases:\n${error.message}",
-                )
-            }
-        }
-    }
+    override fun onRestoreError(error: PurchasesError) {}
```
Comment (NachoSoto) [inline examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/paywall/PaywallScreenViewModel.kt:68]: This is handled by us now.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywall.kt [hunk 2]
```diff
@@ -57,6 +57,13 @@ internal fun InternalPaywall(
 
             is PaywallState.Loaded -> {
                 LoadedPaywall(state = state, viewModel = viewModel)
+
+                viewModel.actionError.value?.let {
+                    ErrorDialog(
+                        dismissRequest = viewModel::clearActionError,
+                        error = it.message,
+                    )
+                }
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywall.kt:66]: Using the new state in `PaywallViewModel`.

**Review by aboedo — APPROVED** [review]

**Review by tonidero — COMMENTED** [review]
I was wondering, have you tried displaying the dialog when presenting the paywall as a dialog? I'm just not sure of the behavior when you try to display a dialog on top of another.

**Comment by NachoSoto** [comment]
@tonidero I did test it, and it does work as expected :)

**Review by tonidero — APPROVED** [review]

**Pull request merged**